File: react/frontend/src/api/contentApi.js
import axios from 'axios';

const apiBaseUrl = 'http://localhost:8020';

export const generateContent = async (prompt, tone, length, audience) => {
  try {
    const response = await axios.post(`${apiBaseUrl}/api/generate_content`, { prompt, tone, length, audience });
    const { generated_content } = response.data;
    return { generated_content };
  } catch (error) {
    console.error('Error generating content:', error.response.data);
    throw error;
  }
};

File: backend/contentApi.py
import requests
from mock_data import mock_content_response

api_base_url = 'http://localhost:5003'

def generate_content(prompt, tone):
    try:
        if api_base_url:
            response = requests.post(f'{api_base_url}/generate_content', json={'prompt': prompt, 'tone': tone})
            return response.json()['generated_content']
        else:
            return mock_content_response['generated_content']
    except Exception as e:
        print('Error generating content:', e)
        raise e